### DISCUSSION TITLE
`addChiralHs` option can't deal with Alanine

### DISCUSSION THREAD

**Discussion by Acylation** [rdkit/rdkit#7231] [Q&A] [answered]
I'm calling rdkit-js in my project, and find this option invalid for Alanine `N[C@@H](C)C(=O)O`. The hydrogen just won't show whether the option is enabled/disabled. Bigger molecules reacts well, e.g. `C1C[C@@H]2C(C)CCCC2CC1`.

Is this behavior the same with python rdkit? Many thanks!

**Comment by ptosco** [comment, marked as answer]
The fact that `addChiralHs` is set to `true` does not mean that a chiral H is going to be added to chiral centers under all circumstances.
With alanine, there is indeed no reason to add a chiral H, since the wedge can be put on the methyl group. For the decalin example, adding an exocyclic chiral H is recommended by [IUPAC guidelines on graphical representation of stereochemical configuration](https://publications.iupac.org/pac/2006/pdf/7810x1897.pdf) (page 1903).
In conclusion, chiral Hs will only added when needed to ensure that the chiral center configuration can be easily perceived by the reader. And, yes, the behavior is the same for all RDKit bindings.

**Reply by Acylation** [reply]
Yeah thank you so much for the quick reply.
